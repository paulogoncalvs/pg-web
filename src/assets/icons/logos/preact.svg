<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 256 296"><path d="M128 0L0 74v148l128 74 128-74V74L128 0zm68 66c11 0 19 3 24 10 12 15 2 43-23 72 25 29 35 57 23 72-14 18-51 11-92-14-41 25-79 32-93 14-12-15-1-43 23-72-24-29-35-57-23-72 5-7 14-10 25-10 18 0 42 9 68 24 25-15 50-24 68-24zm0 9c-7 0-14 2-23 4-11 4-24 9-37 17a298 298 0 0155 45l14-20c11-18 14-32 8-40-3-4-9-6-17-6zM59 75c-8 0-14 2-17 6-6 8-3 22 8 40l15 20a282 282 0 0154-45c-13-8-26-13-37-17-9-2-16-4-23-4zm69 26a284 284 0 00-57 47 272 272 0 0057 47 283 283 0 0056-47 272 272 0 00-56-47zm0 27a20 20 0 010 40 20 20 0 010-40zm-63 27l-15 21c-11 17-14 32-8 39s20 8 40 2c11-4 24-9 37-16a298 298 0 01-54-46zm126 0a282 282 0 01-55 46c13 7 26 12 37 16 20 6 34 5 40-2s3-22-8-39l-14-21z"/></svg>