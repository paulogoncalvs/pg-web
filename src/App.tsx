import { h, FunctionalComponent } from 'preact';
import { Route, Router } from 'wouter-preact';
import { StoreContextProvider } from '@/modules/store';
import { RouterOnChange } from '@/modules/router';
import { getPage } from '@/modules/router/pages';
import { HeadUpdater } from '@/components/HeadUpdater';
import { Header } from '@/components/Header';
import { SideDrawer } from '@/components/SideDrawer';
import { Footer } from '@/components/Footer';
import { Overlay } from '@/components/Overlay';

interface AppProps {
    store?: PageStore;
}

export const App: FunctionalComponent<AppProps> = ({ store = {} }): JSX.Element => (
    <StoreContextProvider store={store}>
        <HeadUpdater />
        <Router ssrPath={store.url}>
            <RouterOnChange />
            <Header />
            <main class="container px-6 pt-16 pb-8 mx-auto sm:pb-16">
                <Route path="*">{(params): JSX.Element => getPage(params['*'] ? `/${params['*']}` : '/')}</Route>
            </main>
            <SideDrawer />
        </Router>
        <Footer />
        <Overlay />
    </StoreContextProvider>
);

export default App;
