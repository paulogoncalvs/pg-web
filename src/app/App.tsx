import { h, FunctionalComponent } from 'preact';
import Router from 'preact-router';
import { AppProvider } from '@/app/AppContext';
import { history, HandleRouterOnChange } from '@/app/router';
import { Header } from '@/components/Header';
import { Footer } from '@/components/Footer';
import { Home } from '@/pages/Home';

export const App: FunctionalComponent<{ store?: PageStore }> = ({ store = {} }): JSX.Element => (
    <AppProvider store={store}>
        <main>
            <Header />
            {/* @ts-ignore */}
            <Router onChange={HandleRouterOnChange} history={history}>
                <Home path="/" />
                <Home path="/:locale" />
            </Router>
        </main>
        <Footer />
    </AppProvider>
);

export default App;
