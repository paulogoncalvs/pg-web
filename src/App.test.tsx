import { h } from 'preact';
import { render } from 'preact-render-to-string';
import { App } from '@/App';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);

describe('App', () => {
    const container = render(<App />);

    it('should render App', async () => {
        expect(container).toMatchSnapshot();
    });

    it('should not have axe violations', async () => {
        const results = await axe(container);

        expect(results).toHaveNoViolations();
    });
});
