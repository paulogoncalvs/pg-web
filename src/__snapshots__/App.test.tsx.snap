// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App 1`] = `"<main><header class=\\"container mx-auto p-4 py-10\\"><div class=\\"flex items-center justify-between\\"><div class=\\"flex items-center\\"><select class=\\"lang-sel text-sm bg-white border-gray-800 hover:bg-gray-300 hover:text-black focus:bg-gray-300 border-2 focus:text-black ml-2 mr-2\\" aria-label=\\"Language selection\\"><option value=\\"en\\" selected>EN</option><option value=\\"pt\\">PT</option></select></div><div class=\\"flex justify-end\\"><a href=\\"https://www.twitter.com/paulogoncalvs\\" class=\\"ic-link _hdr\\" aria-label=\\"Twitter (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"24\\" height=\\"24\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://www.github.com/paulogoncalvs\\" class=\\"ic-link _hdr\\" aria-label=\\"GitHub (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"24\\" height=\\"24\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://pt.linkedin.com/in/paulogoncalvs\\" class=\\"ic-link _hdr\\" aria-label=\\"LinkedIn (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"24\\" height=\\"24\\"><use href=\\"undefinedundefined\\"></use></svg></a></div></div></header><div class=\\"container mx-auto px-6 py-4 sm:py-8\\"><div class=\\"flex flex-col items-center text-center\\"><h1 class=\\"sm:my-2 text-2xl sm:text-5xl font-black animate-fade-in-up-1 opacity-0\\">PAULO GONÃ‡ALVES</h1><h2 class=\\"text-xl sm:text-4xl animate-fade-in-up-2 opacity-0 pb-14 lowercase\\">Front-End Engineer</h2><img width=\\"230\\" height=\\"230\\" alt class=\\"lazy animate-pulse bg-gray-200 rounded-full\\" src=\\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\\" /><div class=\\"py-14 text-center space-y-2 text-sm sm:text-base\\"><p>Born in Vila Nova de Gaia, Portugal.</p><p>Currently working as a Lead Front-End Engineer at Porto Tech Center - Jumia.</p><p>Check the <a href=\\"https://pt.linkedin.com/in/paulogoncalvs\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"underline\\">LinkedIn profile</a> for more information.</p></div><div class=\\"pb-14\\"><h2 class=\\"text-base sm:text-xl font-bold pb-4\\">Wanna reach out?</h2><a href=\\"mailto:contact@paulogoncalves.dev\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"transition transform duration-100 hover:translate-x-1 inline-flex px-4 py-2 sm:py-3 items-center justify-center rounded-full shadow-md bg-gradient-to-r from-green-300 via-green-100 to-green-100 hover:from-green-300 hover:via-green-050 hover:to-green-050\\"><svg aria-hidden=\\"true\\" class width=\\"24\\" height=\\"24\\"><use href=\\"undefinedundefined\\"></use></svg><span class=\\"pl-2 sm:text-xl text-gray-900 dark:text-gray-900\\">E-mail me</span></a></div></div></div></main><footer><div class=\\"bg-gray-100 text-center\\"><div class=\\"container mx-auto px-6 pt-14 pb-12 flex flex-col items-center\\"><div class=\\"pb-6 text-gray-600 text-sm sm:text-base\\">This website was built from scratch with:</div><div class=\\"flex flex-wrap justify-center align-middle\\"><a href=\\"https://www.typescriptlang.org/\\" class=\\"ic-link _ftr\\" aria-label=\\"TypeScript (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://preactjs.com/\\" class=\\"ic-link _ftr\\" aria-label=\\"Preact (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://webpack.js.org/\\" class=\\"ic-link _ftr\\" aria-label=\\"Webpack (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://yarnpkg.com/\\" class=\\"ic-link _ftr\\" aria-label=\\"Yarn (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://eslint.org/\\" class=\\"ic-link _ftr\\" aria-label=\\"ESLint (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://stylelint.io/\\" class=\\"ic-link _ftr\\" aria-label=\\"StyleLint (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://developers.google.com/web/tools/workbox\\" class=\\"ic-link _ftr\\" aria-label=\\"Workbox (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://tailwindcss.com/\\" class=\\"ic-link _ftr\\" aria-label=\\"Tailwind CSS (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://prettier.io/\\" class=\\"ic-link _ftr\\" aria-label=\\"Prettier (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://jestjs.io/\\" class=\\"ic-link _ftr\\" aria-label=\\"Jest (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://postcss.org/\\" class=\\"ic-link _ftr\\" aria-label=\\"PostCSS (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://sass-lang.com/\\" class=\\"ic-link _ftr\\" aria-label=\\"Sass (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"32\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a><a href=\\"https://babeljs.io/\\" class=\\"ic-link _ftr\\" aria-label=\\"Babel (opens in a new window)\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\"><svg aria-hidden=\\"true\\" class=\\"fill-current\\" width=\\"60\\" height=\\"32\\"><use href=\\"undefinedundefined\\"></use></svg></a></div></div></div><div class=\\"container mx-auto py-8 flex flex-col items-center text-center text-gray-500 text-xs\\"><p>optimized for browsers with JavaScript modules support</p><p class=\\"pt-2\\">paulogoncalves.dev Â© 2021  ðŸ™‚</p></div></footer>"`;
